clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -778,9 +778,10 @@ public Future<RecordMetadata> send(ProducerRecord<K, V> record) {
     /**
      * Asynchronously send a record to a topic and invoke the provided callback when the send has been acknowledged.
      * <p>
-     * The send is asynchronous and this method will return immediately once the record has been stored in the buffer of
+     * The send is asynchronous and this method will return immediately (except for getting the topic metadata) once the record has been stored in the buffer of
```

Comment:
@IvanVas, it looks like the code does block waiting for metadata information on the cluster but only if it doesn't have the necessary information on the topic. In other works the producer only blocks on the metadata response, if this is the first record being sent to the cluster by this client for the given topic.

How about encapsulating that in the comment?

Reply:
> @IvanVas, it looks like the code does block waiting for metadata information on the cluster but only if it doesn't have the necessary information on the topic. In other works the producer only blocks on the metadata response, if this is the first record being sent to the cluster by this client for the given topic.
> 
> How about encapsulating that in the comment?

Thanks for the clarification! What do you think about the updated description?